### ISSUE TITLE
How to Manually rotate when certs are expired.

### ISSUE THREAD

**Issue by ibigpapa** [rancher/rke#1487]
**RKE version: v0.1.8-rc11** 

**Docker version: Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 03:35:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 03:35:14 2017
 OS/Arch:      linux/amd64
 Experimental: false
(`docker version`,`docker info` preferred)**

**Operating system and kernel: Ubuntu 16.04.4 LTS,4.4.0-130-generic
 (`cat /etc/os-release`, `uname -r` preferred)**

**Type/provider of hosts:VMWare (VirtualBox/Bare-metal/AWS/GCE/DO)**

**cluster.yml file:
```
ssh_key_path: {PATH}
nodes:
  - address: {IP-1}
    user: root
    role:
      - controlplane
      - etcd
      - worker
  - address: {IP-2}
    user: root
    role:
      - controlplane
      - etcd
      - worker
  - address: {IP-3}
    user: root
    role:
      - controlplane
      - etcd
      - worker
  - address: {IP-4}
    user: root
    role:
      - worker
  - address: {IP-5}
    user: root
    role:
      - worker    
  - address: {IP-6}
    user: root
    role:
      - worker
ignore_docker_version: false
cluster_name: arm
kubernetes_version: "v1.10.3-rancher2-1"
addon_job_timeout: 30
authentication: 
  strategy: "x509"
ingress: 
  provider: "nginx"
network: 
  plugin: "canal"
services: 
  etcd: 
    extra_args: 
      election-timeout: "5000"
      heartbeat-interval: "500"
    snapshot: false
  kube_api: 
    pod_security_policy: false
  kubelet: 
    fail_swap_on: false
    extra_args:
      authentication-token-webhook: true
ssh_agent_auth: false
```
**

**Steps to Reproduce: kubectl version**

**Results:
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:17:28Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate has expired or is not yet valid
```**
Since my version of rancher isn't supported in the 2.0 version of rke and my certs are expired so i can't upgrade the version which means i can't use the cert rotate command.  How would you do this manually?

**Comment by galal-hussein** [comment]
you can still upgrade to the latest version of rke and run the rotation command on the old cluster, for more information you can checkout this blog post explaining the process https://rancher.com/blog/2019/kubernetes-certificate-expiry-and-rotation-in-rancher-kubernetes-clusters

**Comment by ibigpapa** [comment]
I attempted to upgrade the CERT but it failed as it couldn't find them.  Then we attempted to embedd the cert into the cluster.rkestatus file and that also failed to update the cert.

We then manually recreated the certs inside the /etc/kubernetes/ssl directory on all Masters and Workers.  After shutting down the masters and workers then bringing up all the masters we could use kubectl again. Unfortunately no existing pods worked including canal and dns.   We attempted to reset the service account tokens and refresh the pods for those and that yielded mixed results.

Then we upgraded the rancher image from v1.10.3-rancher2-1 to v1.12.9-rancher1-1 using RKE.  This yielded some results.    Still had some pods that didn't work and it appeared the base services for Kubernetes (controller-manager, scheduler, ect) were still reporting certificate issues.

We then used rke to rotate the certs.  Didn't make a difference.  Reset service account tokens again no change.  Attempted to change some of the extra args on various services with no results. Tried both certs and CA with some additional attempts no luck.

Scaled down all deployments, statefulsets to 0.  Then we upgraded from v1.12.9-rancher1-1 to v1.14.3-rancher1-1.  That's when the magic happened all of our base services started working DNS, Controller MGR, Scheduler, Canal, Ingresses.  We then scaled the deployments and stateful sets  back up.  And we had to manually delete all Pods for Daemonsets. 

Fortunately it worked without having to rebuild.  So far so good no major errors we see.  Hope this might come in use for someone in the future.
